clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java

```diff
@@ -360,6 +367,18 @@ void testSendUnsentRequest() {
         assertFalse(networkClient.hasAnyPendingRequests());
     }
 
+    @Test
+    void testMetadataErrorEvent() {
+        metadata.fatalError(new AuthenticationException("Authentication failed"));
+
+        consumerNetworkThread.runOnce();
+        BackgroundEvent event = backgroundEventsQueue.poll();
+        assertNotNull(event);
+        assertEquals(BackgroundEvent.Type.ERROR, event.type());
+        assertEquals(AuthenticationException.class, ((ErrorEvent) event).error().getClass());
+        assertEquals("Authentication failed", ((ErrorEvent) event).error().getMessage());
```

Comment:
this could be simplified to a single `assertEquals(authException, ((ErrorEvent) event).error())`, with authException declared above and passed to the fatalError on ln 372